translation_layer/jpl-capi.js

```diff
@@ -0,0 +1,150 @@
+/*globals console, window*/
+define ('main',['require','jquery','ExtendedModel','api/snapshot/adapters/BackboneAdapter','sim-common/Detect','sim-common/VersionCheck','text!sim-common/templates/notSupported.html','api/snapshot/adapters/BackboneAdapter','api/snapshot/Controller','api/snapshot/Transporter','api/snapshot/SimCapiValue'],function(require){
+
+    var $ = require('jquery');
+    var $body;
+
+    var ExtendedModel = require('ExtendedModel');
+    var adapter = require('api/snapshot/adapters/BackboneAdapter').getInstance();
+
+    var Capi = ExtendedModel.extend({});
+    var capi;
+
+    require('sim-common/Detect');
+    require('sim-common/VersionCheck');
+
+    var notSupportedTemplate = require('text!sim-common/templates/notSupported.html');
+    var BackboneAdapter = require ('api/snapshot/adapters/BackboneAdapter').getInstance();
+    var Controller = require('api/snapshot/Controller');
+    var Transporter = require('api/snapshot/Transporter').getInstance();
+    var Types = require('api/snapshot/SimCapiValue').TYPES;
+
+	var initialized = false;
+
+    window.receiveExposeFromRanger = function (name, type, value, allowedValues)
+    {
+        if (!initialized)
+        {
+            Controller.notifyOnReady();
+            capi = new Capi();
+            capi.on('change', onCapiChange);
+            initialized = true;
+        }
+
+        var typeConverter = generateTypeConverter(type);
+        value = typeConverter(value);
+
+        var descObj = {};
+
+        //console.log("allowed values: " + allowedValues);
+
+        if (allowedValues)
+        {
+            descObj.allowedValues = allowedValues.map(typeConverter);
+        }
+
+        capi.set(name, value);
+        adapter.expose(name, capi, descObj);
+    };
+
+	    window.receiveValueFromRanger = function (name, type, value) {
+        if (initialized) {
+            value = generateTypeConverter(type)(value);
+            capi.set(name, value);
+        }
+    };
+
+    window.UpdateScreenAtSPR = function (_func) 
+    {
+        Transporter.triggerCheck();
+    }
+	
+    function storageSuccess() {
+        console.log("Storage process successful for ");
+        console.log(Transporter.getHandshake().authToken);
+    };
+
+    function storageError() {
+        console.log("Storage process errored!");
+    };
+	
+	function transmitDataToSim(value)
```

Comment:
Very - we are still waiting on API from JPL. The core ideas are what I intended to follow though.